src/main/java/com/example/chapter1/basicprogramming/ReverseArray.java

```diff
@@ -0,0 +1,42 @@
+package com.example.chapter1.basicprogramming;
+
+//TODO: I'm not in favor of static methods. If someone want's make it better please contribute.
+//TODO: I'm not able to make methods generic which accepts primitives. If someone make it better please contribute.
+//TODO: Logic seems to duplicated. Refactor it!!
+public class ReverseArray {
+
+
+    public static int[] reverse(int[] numericArray) {
+        int arrayLength = numericArray.length;
+        if (arrayLength == 0) throw new IllegalArgumentException("Empty array.");
+        for (int index = 0; index < arrayLength / 2; index++) {
+            int temp = numericArray[index];
+            numericArray[index] = numericArray[arrayLength - 1 - index];
+            numericArray[arrayLength - index - 1] = temp;
+        }
+        return numericArray;
+    }
+
+    public static double[] reverse(double[] numericArray) {
```

Comment:
Yeah need to check on this.
If we change the signature from accepting and returning arrays to list it would be possible because of autoboxing but i am not sure for arrays.